test/socket.jl

```diff
@@ -174,3 +174,56 @@ begin
     send(b, ip"::1", port, "Hello World")
     wait(tsk)
 end
+
+begin
+    default_port = UInt16(11011)
+    default_addr = IPv4("127.0.1.1")
+
+    sock = Base.TCPServer()
+    bind(sock,Base.InetAddr(default_addr,default_port))
+    listen(sock)
+
+    new_addr, new_port = getsockname(sock)
+
+    @test default_addr == new_addr
+    @test default_port == new_port
+    close(sock)
+end
+
+@unix_only begin
+    default_port = UInt16(21011)
+    default_addr = IPv6("::1")
+
+    sock = Base.TCPServer()
+    addr = Base.InetAddr(default_addr,default_port)
+    bind(sock,addr)
+    listen(sock)
+
+    new_addr, new_port = getsockname(sock)
+
+    @test default_addr == new_addr
+    @test default_port == new_port
+    close(sock)
+end
+
+begin
+    default_port = UInt16(11011)
+    default_addr = IPv4("127.0.1.1")
+
+    sock = Base.TCPServer()
+    bind(sock,Base.InetAddr(default_addr,default_port))
+    listen(sock)
+
+    @async begin
+        sleep(1)
+        ssock = connect(default_addr, default_port)
+    end
+
+    csock = accept(sock)
+    new_addr, new_port = getsockname(csock)
+
+    @test new_addr == IPv4("127.0.0.1")
```

Comment:
The point of this test to have different addresses for the server and the client. I bind server to  `127.0.1.1`, So when I call `getsockname` on a client socket, accepted by the server, the result should be an address of the connected client, assuming it is `localhost`. See netstat dump below: 
```
tcp        0      0 127.0.1.1:11011         0.0.0.0:*               LISTEN      24146/julia4    
tcp        0      0 127.0.1.1:11011         127.0.0.1:47940         ESTABLISHED 24146/julia4 
```
I pass this test on my machine. Unfortunately, travis thinks differently. Have any Idea?